### ISSUE TITLE
Chapter 9, Case study 1 - no buy signal on the test set

### ISSUE THREAD

**Issue by Abaminog** [tatsath/fin-ml#3]
Hello,

I have been trying to run Case 1 of Chapter 9 (Reinforcement Learning) both on Jupyter and on Pyhton on my local PC. Every time I get a blank result from running the trained model on the test set - there are no buy signals and hence no P&L generated. I tried different options including using the train set instead of the test set, but still the result is the same. The model training goes ok by going through a number of episodes, but then when it comes to using the model that has been created, there seems to be an issue as there are no buy signals generated. Could you please look into this to see if the Jupyter code as posted on GitHub works fine and whether there are any typos or errors in it.

Thank you and kind regards,
Alex M.

**Comment by lishuo520** [comment]
I use the Jupyter code, and get the same result with Alex M. Actions always are 2. I don‘t know if hyperparameters is different.  Could you please check the code.

Thank you!

**Comment by sidaritech** [comment]
Hello, I found a typo or logical error in chapter 9 - RL model page 290. 

Figure 9-4 describes:
In state S1 (stagnant market) there are 2 option:
1) hold a0
2) sell a2

But the paragraph explain another way:
[...]In state s1 it has only two possible actions:
1) hold a0
2) buy a1

Which is correct?
Thank you.

**Comment by darenwai** [comment]
Same here. Using all the library with the same version listed in the requirement. No errors at all but also no result.
If you run it at Binder also same with error your cpu doesnt support the tensorflow instruction.
I have tried this at various pc with the same result.

The book is so well written. I thought the sample would work. Already spent dozen of hours still unable to make it work with my 20 years of c# experience. I am hoping python expert would come in to fix the broken code.

**Comment by tatsath** [comment]
Its quite sensitive to the seed and probably change in python version. We might update it in the next iteration.

**Comment by darenwai** [comment]
Hi. May i know what is the python version when this sample work? I try to install that specific version. Thank you.

**Comment by IGleiser** [comment]
Hello, can you please let us know if you were able to find out which version of Python we need to use?

**Comment by adsd19** [comment]
I had a high hopes using this Chapter#9 Case#1. The training is taking a lots of time, my computer is running the cell since last 120 mins. And now I see that there are issues in the BUY signal as well. It is very disappointing.

**Comment by miladasgari380** [comment]
Hello, I tried a smaller batch size and it fixed the issue for me. My test data was smaller than the tutorial so having a smaller batch size makes sense.

**Comment by darenwai** [comment]
hi miladasgari380.,
 I use a test size of 0.01 and it still does not work. Can i know what is the exact python, tensorflow and keras you are using?

**Comment by rosscortb** [comment]
Am I right in saying that it only opens Long positions?
